Update VolumeSnapshot status for failed snapshots

harsh-px opened the pull request:
Signed-off-by: Harsh Desai <harsh@portworx.com>

Commit:
Update VolumeSnapshot status for failed snapshots

Signed-off-by: Harsh Desai <harsh@portworx.com>

disrani-px reviewed (approved):
LGTM.

bindandUpdateVolumeSnapshot() might need to be renamed since it isn't binding the snapshot object in this case.

Commit:
rename bind snapshot method

Signed-off-by: Harsh Desai <harsh@portworx.com>

harsh-px commented:
> bindandUpdateVolumeSnapshot() might need to be renamed since it isn't binding the snapshot object in this case.

I've renamed it to updateVolumeSnapshot and updated function doc. Can you merge this?
